src/firefly/js/charts/ChartUtil.js

```diff
@@ -652,6 +654,34 @@ export function applyDefaults(chartData={}, resetColor = true) {
     });
 }
 
+/**
+ * Convert column expression to h2 database syntax
+ * @param {string} colOrExpr - column expression
+ * @param {string[]} colNames - valid column names
+ * @returns {*}
+ */
+export function formatColExpr(colOrExpr, colNames) {
+    const expr = new Expression(colOrExpr, colNames);
+    if (expr.isValid()) {
+        // quote columns, assuming column names are alpha-numeric
+        expr.getParsedVariables().forEach( (v) => {
+            const re = new RegExp('([^A-Za-z\d_"]|^)('+v+')([^A-Za-z\d_"]|$)', 'g');
+            colOrExpr = colOrExpr.replace(re, '$1"$2"$3'); // add quotes
+        });
+
+        // substitute functions ceil, lg, ln with the functions h2 understands
```

Comment:
If the decision is to support these functions as is, then the indirection should happen in DbAdapter.  This is where you would convert publicly supported functions into database specific syntax.